scripts/dataset_processing/tts/aishell3/get_data.py

```diff
@@ -0,0 +1,153 @@
+        "--manifests-path", type=Path, help="where the resulting manifests files will reside", default="./"
+    )
+    parser.add_argument("--val-size", default=0.01, type=float, help="eval set split")
+    parser.add_argument("--test-size", default=0.01, type=float, help="test set split")
+    parser.add_argument(
+        "--seed-for-ds-split",
+        default=100,
+        type=float,
+        help="Seed for deterministic split of train/dev/test, NVIDIA's default is 100",
+    )
+
+    args = parser.parse_args()
+    return args
+
+
+def __maybe_download_file(source_url, destination_path):
+    if not destination_path.exists():
+        tmp_file_path = destination_path.with_suffix('.tmp')
+        urllib.request.urlretrieve(source_url, filename=str(tmp_file_path))
+        tmp_file_path.rename(destination_path)
+
+
+def __extract_file(filepath, data_dir):
+    try:
+        tar = tarfile.open(filepath)
+        tar.extractall(data_dir)
+        tar.close()
+    except Exception:
+        print(f"Error while extracting {filepath}. Already extracted?")
+
+
+def __process_transcript(file_path: str):
+    # Create directory for processed wav files
+    Path(file_path / "processed").mkdir(parents=True, exist_ok=True)
+    # Create zh-TW to zh-simplify converter
+    cc = OpenCC('t2s')
+    # Create normalizer
+    text_normalizer = Normalizer(
+        lang="zh", input_case="cased", overwrite_cache=True, cache_dir=str(file_path / "cache_dir"),
+    )
+    text_normalizer_call_kwargs = {"punct_pre_process": True, "punct_post_process": True}
+    normalizer_call = lambda x: text_normalizer.normalize(x, **text_normalizer_call_kwargs)
+    entries = []
+    i = 0
+    SPEAKER_LEN = 7
+    with open(file_path / "train" / "content.txt", encoding="utf-8") as fin:
+        for line in fin:
+            content = line.split()
+            wav_name, text = content[0], "".join(content[1::2]) + "。"
+            wav_name = wav_name.replace(u'\ufeff', '')
+            speaker = wav_name[:SPEAKER_LEN]
+            wav_file = file_path / "train" / "wav" / speaker / wav_name
+            assert os.path.exists(wav_file), f"{wav_file} not found!"
+            duration = subprocess.check_output(f"soxi -D {wav_file}", shell=True)
+            if float(duration) <= 3.0:  # filter out wav files shorter than 3 seconds
+                continue
+            processed_file = file_path / "processed" / wav_name
+            # convert wav to mono 22050HZ, 16 bit (as SFSpeech dataset)
+            subprocess.run(f"sox {wav_file} -r 22050 -c 1 -b 16 {processed_file}", shell=True)
+            simplified_text = cc.convert(text)
+            normalized_text = normalizer_call(simplified_text)
+            entry = {
+                'audio_filepath': os.path.abspath(processed_file),
+                'duration': float(duration),
+                'text': text,
+                'normalized_text': normalized_text,
+                'speaker': int(speaker[3:]),
```

Comment:
double check `speaker` value according to my previous comment.

Reply:
As above, I used last 4 digits as speaker_id. For example, if the speaker is `SSB1902` in AISHELL3, the speaker_id will be `1902`